piezo_web_app/PiezoWebApp/src/services/spark_job/spark_job_service.py

```diff
@@ -81,6 +81,11 @@ def get_jobs(self):
             spark_jobs = {
                 item['metadata']['name']: SparkJobService._retrieve_status(item) for item in api_response['items']
             }
+            if label.upper() != "ALL":
+                for item in api_response['items']:
+                    user_label = SparkJobService._get_user_label(item)
+                    if user_label != label:
+                        del spark_jobs[item['metadata']['name']]
```

Comment:
Is it not possible to give the required label when calling the kubernetes adapter?